packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```diff
@@ -0,0 +1,355 @@
+    send(event: ITelemetryBaseEvent): void {
+        if (this.ignoredEvents.has(event.eventName)) {
+            let matches = true;
+            const ie = this.ignoredEvents.get(event.eventName);
+            assert(ie !== undefined);
+            for (const key of Object.keys(ie)) {
+                if (ie[key] !== event[key]) {
+                    matches = false;
+                    break;
+                }
+            }
+            if (matches) {
+                event.category = "generic";
+            }
+        }
+
+        this.baseLogger.send(event);
+    }
+}
+
+/**
+ * Validates this scenario: When a datastore is aliased that it is considered a root datastore and always referenced
+ */
+describeNoCompat("GC Random tests", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    const dataObjectFactory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            summaryConfigOverrides: {
+                state: "disableHeuristics",
+                initialSummarizerDelayMs: 0,
+                summarizerClientElection: false,
+                maxAckWaitTime: 100,
+                maxOpsSinceLastSummary: 100,
+            },
+        },
+        gcOptions: {
+            gcAllowed: true,
+        },
+    };
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        runtimeOptions,
+    );
+
+    const defaultSessionExpiryDurationMs = 3000; // 3 seconds
+
+    // Enable config provider setting to write GC data at the root.
+    const settings = {
+        "Fluid.GarbageCollection.RunSessionExpiry": "true",
+        "Fluid.GarbageCollection.TestOverride.SessionExpiryMs": defaultSessionExpiryDurationMs,
+    };
+    const configProvider = mockConfigProvider(settings);
+
+    const uploadedSummaries: ISummaryTree[] = [];
+
+    let overrideLogger: IgnoreErrorLogger;
+
+    let mainContainer: IContainer;
+    let connectedContainers: IContainer[];
+    let closedContainers: IContainer[];
+    let testNodes: Map<string, TestNode>;
+    let unrefDataStores: Set<string>;
+    let refDataStores: Set<string>;
+    let rootDataStores: Set<string>;
+
+    const createContainer = async (): Promise<IContainer> => {
+        return provider.createContainer(runtimeFactory, { configProvider });
+    };
+    const loadContainer = async (): Promise<IContainer> => {
+        return provider.loadContainer(runtimeFactory, { configProvider });
+    };
+
+    const sleep = async (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
```

Comment:
Something to ponder - Do we actually need real time to pass or can we use Sinon to mock the clock too?

Reply:
Sinon starts getting weird when you call provider.ensureSynchronized(). I'm not sure we actually need to call it, I'll consider it.